tests/Integration/Behaviour/Features/Context/Domain/OrderReturnFeatureContext.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Configuration;
+use Order;
+use OrderReturn;
+use OrderReturnState;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Command\UpdateOrderReturnStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnConstraintException;
+use PrestaShopDatabaseException;
+use PrestaShopException;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+
+class OrderReturnFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add order return state :orderReturnStateReference
+     *
+     * @param string $orderReturnStateReference
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function createOrderReturnState(string $orderReturnStateReference): void
```

Comment:
Since these statements are mainly used to created fixtures for your behat tests I don't think they should be in this Domain context
Usually we separate these context in the "upper" namespace, a bit like we did for Order:

Context to manipulate legacy objects directly (for features or low levels checks)
https://github.com/PrestaShop/PrestaShop/blob/ee00c133ac5140a427f45b6ea5d948cb34f9123b/tests/Integration/Behaviour/Features/Context/OrderFeatureContext.php 

Context that uses CQRS domain commands:
https://github.com/PrestaShop/PrestaShop/blob/22d5e6b339f481423c89678c065f5681833c11f8/tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

Reply:
So `createOrderReturnState` and `createOrderReturn` should be moved in a different context But you can still use `SharedStorage::getStorage()` as a common "proxy"